src/toolbar/Toolbar.ts

```diff
@@ -0,0 +1,158 @@
+import { debounce } from '@dojo/core/util';
+import { Dimensions } from '@dojo/widget-core/meta/Dimensions';
+import { DNode, WidgetProperties } from '@dojo/widget-core/interfaces';
+import { ThemedMixin, theme } from '@dojo/widget-core/mixins/Themed';
+import { v, w } from '@dojo/widget-core/d';
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+
+import SlidePane, { Align } from '../slidepane/SlidePane';
+import * as css from './styles/toolbar.m.css';
+import * as iconCss from '../common/styles/icons.m.css';
+
+/**
+ * @type ToolbarProperties
+ *
+ * Properties that can be set on a Toolbar component
+ *
+ * @property actions           Action elements to show in the toolbar
+ * @property collapseWidth     Width at which to collapse actions into a SlidePane
+ * @property fixed             Fixes the toolbar to the top of the viewport
+ * @property title             Element to show as the toolbar title
+ */
+export interface ToolbarProperties extends WidgetProperties {
+	actions?: DNode[];
+	collapseWidth?: number;
+	fixed?: boolean;
+	title?: DNode;
+};
+
+export const ThemedBase = ThemedMixin(WidgetBase);
+
+@theme(css)
+@theme(iconCss)
+export default class Toolbar extends ThemedBase<ToolbarProperties> {
+	private _collapsed = false;
+	private _onResize: () => void;
+	private _open = false;
+
+	constructor() {
+		super();
+		this._onResize = debounce(this._collapseIfNecessary, 250);
+		window.addEventListener('resize', this._onResize);
```

Comment:
This is stopgap until dojo/widget-core#618 lands. For now, we do the best we can by debouncing a resize handler that's explicitly removed on detachment.